### PR TITLE
682 update imported data from repos

### PR DESCRIPTION
**Pull request by mohammad-sdsc**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

# Description

Provides an update mechanism for datasets that do not use submodules. One can run `renku dataset update` to update all such datasets and can use `-I` and `-X` to include and exclude specific paths or specify dataset names as argument. Run `renku dataset update -h` for more info.

Fixes https://github.com/SwissDataScienceCenter/renku-python/issues/682

## Type of change

Please select relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

**Do not create pull request unless you checked all points.**

- [ ] I have read and understood the **CONTRIBUTING** document.
- [ ] I have performed a self-review of my own code.
- [ ] I have signed and sent the [CLA document]
      (https://github.com/SwissDataScienceCenter/documentation/wiki/files/SDSC_Contributor_License_Agreement_v1.0.pdf).
- [x] I have added tests to cover my changes.
- [x] I have **NOT** removed any existing tests.
- [ ] I have updated the documentation accordingly.
- [ ] I have described *all* changes in the **CHANGELOG.rst**.
- [x] I have run ``./run-tests.sh`` locally.

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -217,7 +217,7 @@ def request_callback(request):
         yield rsps
 
 
-@pytest.fixture(scope='module')
+@pytest.fixture(scope='function')
```
Comment (jsam) [inline conftest.py]: Why is this needed?

File: renku/cli/dataset.py [hunk 2]
```diff
@@ -619,3 +619,31 @@ def _init(lock, id_queue):
...
+    help='Include files matching given pattern.'
+)
+@click.option(
+    '-X',
+    '--exclude',
+    default=None,
+    multiple=True,
+    help='Exclude files matching given pattern.'
+)
+def update(names, creators, include, exclude):
```
[the remark this answers is not in the document]
Comment (mohammad-sdsc) [reply renku/cli/dataset.py]: I would say keep it simple for now. Users have `-X` to exclude some files. Moreover, if something is deleted from the remote repo then they have to put `--no-delete` in each update command. Maybe it's just easier for them to delete the file and add it locally (to remove the link to the remote).

**Comment by rokroskar** [comment]
When there is nothing to update, this creates an empty commit - is this desired?
